python/py-htseq/Portfile

```diff
@@ -16,16 +16,16 @@ description         A framework to process and analyze data from high-throughput
                     sequencing (HTS) assays
 long_description    ${description}
 
-homepage            https://github.com/simon-anders/[string tolower ${_name}]
+homepage            https://github.com/htseq/[string tolower ${_name}]
 master_sites        pypi:${_n}/${_name}
 distname            ${_name}-${version}
 
-checksums           rmd160  a86dc2c644eb4058c7bdc0b47fdbc5a024605412 \
-                    sha256  65c4c13968506c7df92e97124df96fdd041c4476c12a548d67350ba8b436bcfc \
-                    size    611204
+checksums           rmd160  6f0c0b89e03cbcfea2bc8a45bc01bddf0d3ea49f \
+                    sha256  0bdc778358e5bd06901639797adb27bf4e85602d621062b62d6be9fee70e645e \
+                    size    625810
 
-# Python 3.8 is not currently supported by dependency py-pysam
-python.versions     27 36 37
+# Python 3.9 is not currently supported by dependency py-pysam
+python.versions     27 36 37 38
 
 if {${name} ne ${subport}} {
     depends_build-append \
```

Comment:
`Cython` is also listed under `setup_requires` by [upstream](https://github.com/htseq/htseq/blob/e96d2214410a9f1bb6475cab5372f2e4efd0ff1c/setup.py#L35).

Reply:
According to README.md, Cython is required to develop HTSeq and not to use it, and the PyPI package does not depend on Cython.
However, py-cython is already installed as a dependency of py-numpy, so I added it.

Reply:
that's not completely true as it is trying to use `cython` during the build, but cannot find it...
```
cythonizing
/bin/sh: cython: command not found
Cython not found, but transpiled file found
```

it might work without and, if so, you can remove the `py-cython` dependency. Otherwise you should list it as a `depends_build` and patch `setup.py` so that it will use the correct Cython version (likely `cython-${python.branch}`.

Reply:
I had a closer look at `setup.py`.
It is true that the build process tries to use Cython and SWIG.
If they are not found it tries to find the files Cython and SWIG would produce: if they are found it continues the build, otherwise it raises an exception.
When building this version of the port, Cython and SWIG are not found, but the PyPI source package indeed contains the required files and the build can continue.

Given that the author(s) explicitly state that Cython and SWIG should not be required to use HTSeq but only to develop it, rather than patching the source to use MacPorts' Cython and SWIG, I decided for now to remove the added dependency and rely on the pre-generated files (note that these are still source code and not binary files).
(For the record, I would not oppose the other approach, should anyone implement it.)

Reply:
okay, fair enough and sorry for the noise ;)